c++/src/capnp/layout.c++

```diff
@@ -30,26 +30,47 @@
 #include "capability.h"
 #endif  // !CAPNP_LITE
 
+#ifdef __has_feature
+#define IS_BUILDING_WITH_TSAN() __has_feature(thread_sanitizer)
+#else
+#if defined(__SANITIZE_THREAD__) && __SANITIZE_THREAD__
+#define IS_BUILDING_WITH_TSAN() 1
+#else
+#define IS_BUILDING_WITH_TSAN() 0
+#endif
+#endif
+
+
 namespace capnp {
 namespace _ {  // private
 
 #if !CAPNP_LITE
-static BrokenCapFactory* brokenCapFactory = nullptr;
+static BrokenCapFactory* gBrokenCapFactory = nullptr;
 // Horrible hack:  We need to be able to construct broken caps without any capability context,
 // but we can't have a link-time dependency on libcapnp-rpc.
 
 void setGlobalBrokenCapFactoryForLayoutCpp(BrokenCapFactory& factory) {
   // Called from capability.c++ when the capability API is used, to make sure that layout.c++
   // is ready for it.  May be called multiple times but always with the same value.
 #if __GNUC__ || defined(__clang__)
-  __atomic_store_n(&brokenCapFactory, &factory, __ATOMIC_RELAXED);
+  __atomic_store_n(&gBrokenCapFactory, &factory, __ATOMIC_RELAXED);
 #elif _MSC_VER
-  *static_cast<BrokenCapFactory* volatile*>(&brokenCapFactory) = &factory;
+  *static_cast<BrokenCapFactory* volatile*>(&gBrokenCapFactory) = &factory;
 #else
 #error "Platform not supported"
 #endif
 }
 
+static BrokenCapFactory* readGlobalBrokenCapFactoryForLayoutCpp() {
+#if IS_BUILDING_WITH_TSAN()
```

Comment:
Let's just have it always be atomic and skip the TSAN check. No need to comment about TSAN either -- it's reasonable enough that if the global is atomically written, it ought to be atomically read.

Reply:
Good call.